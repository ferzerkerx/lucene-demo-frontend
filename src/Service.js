import {checkStatus, parseJSON} from './Utils'

export const Service = (text) => {
    const encodedQuery = encodeURIComponent(text);
    return fetch(`http://localhost:8080/books?keyword=${encodedQuery}`, {credentials: 'include'})
        .then(checkStatus)
        .then(parseJSON)
        .catch((error)  => console.error('request failed', error));
};