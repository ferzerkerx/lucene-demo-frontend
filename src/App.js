import React, { Component } from 'react';
import './App.css';
import { Service } from './Service';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = { results: undefined };
  }

  handleOnChange = evt => {
    evt.preventDefault();
    const text = evt.target.value;
    Service(text).then(results => {
      this.setState({ results: results });
    });
  };

  render() {
    const renderedResults = this.renderedResults();
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">
            Auto complete input backed by an endpoint using Lucene
          </h1>
        </header>
        <p className="App-intro">Type any book you want.</p>
        <form onSubmit={e => e.preventDefault()}>
          <input type="text" onChange={this.handleOnChange} />
        </form>
        <div className="results">{renderedResults}</div>
      </div>
    );
  }

  renderedResults() {
    const results = this.state.results;
    return (
      results &&
      results.map(book => {
        return (
          <div className="book-result">
            <span className="book-title">{book.title}</span> by{' '}
            <span className="author-name">{book.author}</span>
          </div>
        );
      })
    );
  }
}

export default App;
