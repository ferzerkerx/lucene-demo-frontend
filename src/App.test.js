import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { Service } from './Service';

jest.mock('./Service', () => ({
  Service: jest.fn(text => Promise.resolve([]))
}));

describe('App.js', () => {
  it('should render component', () => {
    const app = shallow(<App />);
    expect(app).toMatchSnapshot();
  });

  it('should handle onChangeEvent', () => {
    const app = shallow(<App />);
    expect(app.state().results).toBeNull();

    const searchInput = app.find("input[name='searchInput']");
    const event = {
      target: { name: 'searchInput', value: 'bookToSearch' },
      preventDefault: () => {}
    };
    searchInput.simulate('change', event);
    expect(Service).toBeCalledWith('bookToSearch');
  });

  it('should render with results', () => {
    const app = shallow(<App />);
    app.setState({
      results: [{ id: 1, title: 'bookTitle', author: 'authorName' }]
    });
    app.update();
    expect(app).toMatchSnapshot();
  });
});
