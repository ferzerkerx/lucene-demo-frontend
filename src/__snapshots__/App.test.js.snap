// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js should render component 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Auto complete input backed by an endpoint using Lucene
    </h1>
  </header>
  <p
    className="App-intro"
  >
    Type any book you want.
  </p>
  <form>
    <input
      name="searchInput"
      onChange={[Function]}
      type="text"
    />
  </form>
  <div
    className="results"
  />
</div>
`;

exports[`App.js should render with results 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Auto complete input backed by an endpoint using Lucene
    </h1>
  </header>
  <p
    className="App-intro"
  >
    Type any book you want.
  </p>
  <form>
    <input
      name="searchInput"
      onChange={[Function]}
      type="text"
    />
  </form>
  <div
    className="results"
  >
    <div
      className="book-result"
      key="1"
    >
      <span
        className="book-title"
      >
        bookTitle
      </span>
       by
       
      <span
        className="author-name"
      >
        authorName
      </span>
    </div>
  </div>
</div>
`;
